spec/lib/phobos_db_checkpoint/failure_spec.rb

```diff
@@ -248,4 +248,20 @@ def event_version(payload)
       end
     end
   end
+
+  describe '.ascending_order' do
+    before do
+      PhobosDBCheckpoint::Failure.delete_all
+      PhobosDBCheckpoint::Failure.create(id: '1', entity_id: '1', created_at: Time.now-100, event_time: nil)
+      PhobosDBCheckpoint::Failure.create(id: '2', entity_id: '2', created_at: Time.now-200, event_time: nil)
+      PhobosDBCheckpoint::Failure.create(id: '3', entity_id: '3', created_at: Time.now-300, event_time: Time.now-300)
+      PhobosDBCheckpoint::Failure.create(id: '4', entity_id: '4', event_time: nil).tap { |r| r.created_at=nil; r.save! }
+      PhobosDBCheckpoint::Failure.create(id: '5', entity_id: '5', event_time: Time.now-400).tap { |r| r.created_at=nil; r.save! }
+      PhobosDBCheckpoint::Failure.create(id: '6', entity_id: '6', event_time: Time.now-500).tap { |r| r.created_at=nil; r.save! }
+    end
+
+    it 'sorts it in ascending order leaving null timestamp records trailing at the end' do
+      expect(PhobosDBCheckpoint::Failure.ascending_order.pluck('entity_id')).to eq ['3', '5', '6', '1', '2', '4']
```

Comment:
This order is really weird for me =(
It seems it's sorting first by `event_time`, and then by `created_at`. While what I would expect is: sort by `event_time`, if you don't have it, use the `created_at`, if you don't have any, put in the end.
So the final `ascending` order (older first) would be:
`%w(6 5 3 2 1 4)`
and the `descending` (older last):
`%w(1 2 3 5 6 4)`